### ISSUE TITLE
Unary Ufunc tracking issue

### ISSUE THREAD

**Issue by mruberry** [pytorch/pytorch#42515]
Tracking issue for tasks related to unary "[universal functions](https://numpy.org/doc/stable/reference/ufuncs.html?highlight=ufunc)," elementwise functions of a single tensor input.

**Tasks**
- [ ] Alias rad2deg and degrees
- [ ] Alias deg2rad and radians
- [ ] torch.real and np.real are divergent on non-complex dtypes
- [ ] torch.imag and np.imag are divergent on non-complex dtypes
- [ ] Alias conj and conjugate
- [ ] Alias bitwise_not and invert
- [ ] Add integral type support to ceil, floor, round, and trunc
- [x] Alias mvlgamma and multigammaln

cc @ezyang @gchanan @zou3519 @bdhirsh @jbschlosser @anjali411 @mruberry @rgommers @heitorschueroff

**Comment by rgommers** [comment]
For `torch.sinc`, please see https://github.com/numpy/numpy/pull/12924 regarding implementation details - the current version in numpy 1.19 isn't very good (error near 0 increases linearly).

**Comment by hameerabbasi** [comment]
I've picked up `torch.degrees` in #42910.

**Comment by hameerabbasi** [comment]
I've picked up `torch.fix` in #43326

**Comment by kshitij12345** [comment]
Will be picking up `torch.exp2`. Will send a PR soon.

**Comment by hameerabbasi** [comment]
Is there a simple example of 2-output ufuncs? I tried to find `divmod` but that doesn't exist either.

**Comment by mruberry** [comment]
> Is there a simple example of 2-output ufuncs? I tried to find `divmod` but that doesn't exist either.

That's true. Actually divmod is another NumPy function we're missing. An example of a PyTorch operation returning two tensors is max. See it's implementation here:

https://github.com/pytorch/pytorch/blob/8acce55015db3ffd0f07a64341f6b7fec0a1e47d/aten/src/ATen/native/TensorCompare.cpp#L286

**Comment by kshitij12345** [comment]
Picking `nan_to_num`.

**Comment by hameerabbasi** [comment]
Is there an easy way to do different implementations for different `dtype`s? See the draft PR. Neither me nor @antocuni have been successful in getting that to work.

This is needed for the different implementation of `sinc` for reals (see https://github.com/pytorch/pytorch/issues/42515#issuecomment-671627496) vs complex numbers.

**Comment by kshitij12345** [comment]
How about?
https://github.com/pytorch/pytorch/blob/b85568a54a9c60986235ad1e0cc5dffc71b9d5b1/aten/src/ATen/native/cuda/UnaryOpsKernel.cu#L68-L88

**Comment by kshitij12345** [comment]
Promote Integer to Float Operators

* [ ] torch.angle
* [ ] torch.exp
* [x] torch.log #46810
* [x] torch.log2 #46810
* [x] torch.log10 #46810
* [ ] torch.expm1
* [ ] torch.log1p
* [ ] torch.sqrt
* [ ] torch.reciprocal (this would be divergent from NumPy, but that's OK)
* [x] torch.sin #45733
* [ ] torch.cos #46706
* [ ] torch.tan #46706
* [ ] torch.asin
* [ ] torch.acos
* [ ] torch.atan
* [ ] torch.sinh
* [ ] torch.cosh
* [ ] torch.tanh
* [ ] torch.erf
* [ ] torch.erfc
* [ ] torch.erfinv
* [ ] torch.rsqrt
* [ ] torch.sigmoid
* [ ] torch.digamma
* [ ] torch.lgamma
* [ ] torch.polygamma
* [ ] torch.mvlgamma

**Comment by vadimkantorov** [comment]
Tangential question: Is torch.sigmoid supposed to be called by users? I thought that only F.sigmoid was a public API

**Comment by mruberry** [comment]
> Tangential question: Is torch.sigmoid supposed to be called by users? I thought that only F.sigmoid was a public API

Yes. It is documented [here](https://pytorch.org/docs/master/generated/torch.sigmoid.html?highlight=sigmoid#torch.sigmoid).

**Comment by mruberry** [comment]
Closing this issue as it's largely been superceded
